docs/src/security/waf.md

```diff
@@ -0,0 +1,145 @@
+
+{{< note title="Protection rules">}}
+The WAF monitors for requests that include the carriage return (CR; `%0d`) and line feed (LF; `%0a`) characters. The presence of these characters indicate an attempt to inject data in the response header and potentially force intermediate proxy servers to treat the message as two separate responses. 
+{{< /note >}}
+
+### HTTP Splitting
+
+[Need some help with intro description here. Seems like much of the above, only specific to filename POSTs. Or Rewrites, but that seems specific to Apache/PHP? Not really sure.]
+
+{{< note title="Protection rules">}}
+The WAF monitors for requests that include `\n` or `\r` characters in the `REQUEST_FILENAME` rewrite variable.
+{{< /note >}}
+
+## HTTP protocol enforcement
+
+On top of the above ruleset, the Platform.sh WAF implements a number of additional rules intended to enforce the HTTP protocol. 
+
+- **HTTP Request Line Format Enforcement**
+
+    Request lines are validated against the HTTP RFC specification using rule negation with regex, which specifies the proper construction of URI request lines (such as `"http:" "//" host [":" port] [ abs_path ["?" query]]`). The Gateway proxy terminates the incoming connection and then reconstructs requests on the internal network, enforcing the valid format in transit.
+
+- **GET or HEAD requests with a body are not allowed**
+
+    The protocol allows for `GET` requests to have a body - while this is rarely used, attackers could try to force a request body on unsuspecting applications. When the Gateway proxy detects a `GET` request, it will check for the existence of the `Content-Length` and `Transfer-Encoding` headers. If either exist and the payload is either not 0 or not empty, then all body-related headers are stripped from those requests and then passed through otherwise unaltered. 
+
+- **Disallow Content-Length and Transfer-Encoding Headers together**
+
+    Requests [must not contain](https://tools.ietf.org/html/rfc7230#section-3.3.2) both a `Content-Length` and `Transfer-Encoding` header, as the presence of leaves applications vulnerable to [request smuggling](#request-smuggling). The WAF forces requests to use chunked transfer-encoding only on the internal network. 
+
+- **Missing/Empty Host Headers**
+
+    The Gateway proxy implements route mapping based on host names. Because of this, requests are not relayed that do either do not include the host header, or if that header exists but is empty. 
+
+- **File upload limit**
+
+    File upload limits are enforced by the Gateway proxy, but are configured by the application's configuration in `.platform.app.yaml` using the `max_requst_size` attribute in [`web.locations`](/configuration/app/web.md#locations).
+
+    [Do we set some default here if not set? Some default limit on multipart/form-data related to https://docs.platform.sh/development/troubleshoot.html#large-file-upload-failing-10mb-limit?]
```

Comment:
clarification I thought of if we know the answer here

Reply:
(File upload limit) users can set max_request_size to configure this, but what is it if unset (10MB like https://docs.platform.sh/development/troubleshoot.html#large-file-upload-failing-10mb-limit?)